core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/TestRewriteCaseExpressionPredicate.java

```diff
@@ -0,0 +1,266 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.planner.iterative.rule;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.metadata.Metadata;
+import io.trino.security.AllowAllAccessControl;
+import io.trino.spi.type.Type;
+import io.trino.sql.ExpressionUtils;
+import io.trino.sql.PlannerContext;
+import io.trino.sql.planner.DomainTranslator;
+import io.trino.sql.planner.Symbol;
+import io.trino.sql.planner.SymbolAllocator;
+import io.trino.sql.planner.TestingPlannerContext;
+import io.trino.sql.planner.TypeProvider;
+import io.trino.sql.planner.iterative.rule.test.BaseRuleTest;
+import io.trino.sql.planner.iterative.rule.test.PlanBuilder;
+import io.trino.sql.tree.Expression;
+import io.trino.transaction.TestingTransactionManager;
+import org.testng.annotations.Test;
+
+import java.util.Map;
+
+import static io.trino.SessionTestUtils.TEST_SESSION;
+import static io.trino.SystemSessionProperties.OPTIMIZE_CASE_EXPRESSION_PREDICATE;
+import static io.trino.metadata.MetadataManager.createTestMetadataManager;
+import static io.trino.spi.type.IntegerType.INTEGER;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static io.trino.sql.planner.TestingPlannerContext.plannerContextBuilder;
+import static io.trino.sql.planner.TypeAnalyzer.createTestingTypeAnalyzer;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.filter;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.values;
+import static io.trino.sql.planner.iterative.rule.SimplifyExpressions.rewrite;
+import static io.trino.transaction.TransactionBuilder.transaction;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestRewriteCaseExpressionPredicate
+        extends BaseRuleTest
+{
+    private static final Metadata METADATA = createTestMetadataManager();
+    private static final PlannerContext PLANNER_CONTEXT = plannerContextBuilder().withMetadata(METADATA).build();
+    private static final DomainTranslator DOMAIN_TRANSLATOR = new DomainTranslator(PLANNER_CONTEXT);
+    private static final Map<Symbol, Type> TYPE_MAP = ImmutableMap.of(new Symbol("col1"), INTEGER, new Symbol("col2"), INTEGER, new Symbol("col3"), VARCHAR);
+
+    private void assertRewriteDoesNotFire(String expression)
+    {
+        tester().assertThat(new RewriteCaseExpressionPredicate(tester().getPlannerContext(), tester().getTypeAnalyzer()))
+                .setSystemProperty(OPTIMIZE_CASE_EXPRESSION_PREDICATE, "true")
+                .on(p -> p.filter(PlanBuilder.expression(expression), p.values()))
+                .doesNotFire();
+    }
+
+    private void assertRewrittenAndOptimizedExpression(String inputExpressionStr,
```

Comment:
Please move all the helper methods below the tests, and reformat to either have all arguments on the same line or all arguments on separate lines. Also, please do not abbreviate. I think that `inputExpression`, `expectedRewritten`, `expectedOptimized` are sufficiently clear names.

Reply:
Done.